src/blas/backends/cublas/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+#==========================================================================
+#  Copyright (C) Codeplay Software Limited
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  For your convenience, a copy of the License has been included in this
+#  repository.
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#=========================================================================
+
+set(LIB_NAME onemkl_blas_cublas)
+set(LIB_OBJ ${LIB_NAME}_obj)
+find_package(cuBLAS REQUIRED)
+find_package(OpenCL REQUIRED)
```

Comment:
Regarding this OpenCL library requirement, looks like the standard cmake config file for OpenCL is looking for OpenCL in very limited set of environment variables and locations. So even if I have LLVM compiler installed on my machine (and it includes OpenCL) this find_package will fail, as the result I have to set cmake variables explicitly to make it work. Do we really need this find-package if we don't work with OpenCL library directly and "clang++ -fsycl" command puts proper header and library from LLVM? 

I've tested it locally, build works fine without 'find_package(OpenCL REQUIRED)'. 
However I have another problem that I see at runtime: it is a conflict between OpenCL libraries from LLVM build and CUDA. Sourcing CUDA first helped me to resolve this conflict. I guess we need some instruction for users what to do if they will face with similar problem.

Reply:
@mkrainiuk , the `find_package(OpenCL REQUIRED)` is extra and is not needed. It was leftover from the rebasing process. The only thing needed is OpenCL headers and I added a path to get it from llvm install folder. I will remove that.

Reply:
Thanks @mehdi-goli! So I think the only one question I have is what the best way to resolve the conflict between OpenCL library from LLVM and OpenCL library form CUDA package. Should we just recommend to source CUDA before LLVM, or there is better way how to resolve it?

Reply:
For now, we need to avoid OpenCL from CUDA. OpenCL from CUDA is based on OpenCL 1.1, so both CUDA OpenCL Heades and the library are outdated. So for now, the best way is to avoid CUDA OpenCL and always link to the Khronos ICD loader for lib OpenCL. and OpenCL headers. (Both are shipped by LLVM compiler at the moment).

Reply:
Agree. But I guess I have one more problem in this case. If I'm using latest ICD loader and header files from LLVM, then I see that OpenCL loads libnvidia-opencl.so and my test fails with CL_INVALID_VALUE at queue creation step. But if I'm using OpenCL library from CUDA package it doesn't download this library and works fine. Could it be that it's only my machine problem or it's common problem?

Reply:
The reason that you see this issue is because probably you have `nvidia.icd` inside `/etc/OpenCL/vendor/` folder, So the Nvidia is picked up by the ICD Loader and causes the issue. If you remove the `nvidia.icd from the ``/etc/OpenCL/vendor/`  you won't see the error. 
The selection of `nvidia.icd` by the runtime is a known issue for LLVM CUDA backend and the codeplay runtime team is working on that to fix it. 

The `CL_INVALID_VALUE` happens as soon as any SYCL kernel picked by Nvidia OpenCL. the cold cache method in the benchmark manifests the issue if Nvidia openCL used. Or any other SYCL kernel written on NVIDIA backend will cause this issue.

Reply:
Thank you! Yes, removing nvidia.icd helped me. Do you know if there is a link to this issue? So we can refer to it if someone will report similar problem?

Reply:
I can figure out. But we can add it as a known issue in the `readme.md`

Reply:
@mkrainiuk I think this is the issue https://github.com/intel/llvm/issues/1240

Reply:
I see the same error message. Great, thank you!